include/zephyr/drivers/counter.h

```diff
@@ -431,6 +433,27 @@ static inline int z_impl_counter_get_value_64(const struct device *dev,
 	return api->get_value_64(dev, ticks);
 }
 
+/**
+ * @brief Reset the counter to the initial value.
+ * @param dev Pointer to the device structure for the driver instance.
+ *
+ * @retval 0 If successful.
+ * @retval Negative error code on failure resetting the counter value.
+ */
+__syscall int counter_reset(const struct device *dev);
+
+static inline int z_impl_counter_reset(const struct device *dev)
+{
+	const struct counter_driver_api *api =
+				(struct counter_driver_api *)dev->api;
+
+	if (!api->reset) {
+		return -ENOTSUP;
```

Comment:
As per https://docs.zephyrproject.org/latest/develop/api/design_guidelines.html#return-codes:

```suggestion
		return -ENOSYS;
```

Reply:
Yes, I know about it, but other APIs do the same thing. Should I fix them as well? I'm concerned this might cause some unknown side effects. E.g.:
```
static inline int z_impl_counter_get_value_64(const struct device *dev,
					   uint64_t *ticks)
{
	const struct counter_driver_api *api =
				(struct counter_driver_api *)dev->api;

	if (!api->get_value_64) {
		return -ENOTSUP;
	}

	return api->get_value_64(dev, ticks);
}
```